onnx/defs/nn/defs.cc

```diff
@@ -1248,6 +1256,44 @@ void convTransposeShapeInference(InferenceContext& ctx) {
     }
   }
 
+  std::vector<int64_t> pads;
+  if (getRepeatedAttribute(ctx, "pads", pads)) {
```

Comment:
I think this if condition should be (parse the auto_pad attribute prior to this like in line 1266)- 

`(nullptr == auto_pad_attr) || (nullptr != auto_pad_attr && auto_pad_attr->s() == "NOTSET" )`

Probably it is safe to also ensure both `auto_pad` and `pads` attributes are not present in this node - if both are detected fail shape inference with appropriate message

Reply:
Since auto_pad is deprecated attribute, "pads" has higher priority than auto_pad and should be checked first. If "pads" and "NOTSET" are not present, it means no padding.